acra/configuring-maintaining/storage-and-data-model-implications/_index.md

```diff
@@ -2,3 +2,79 @@
 title: Storage and data model implications
 bookCollapseSection: true
 ---
+
+# Data storage
+
+Based on which Acra components you use, the requirements may differ.
+
+For example, you will most likely want to use AcraServer.
+In this case, you are limited by supported DBMS, which are MySQL, MariaDB, PostgreSQL.
+
+On the other hand, there is AcraTranslator and it does not depend on storage you use.
+It's just a side service that performs crypto operations and you are free to store encrypted data in
+any other database (including embedded ones), in any key/value storage and so on.
+The main requirement for a storage would be support of binary fields (value as array of bytes, blob, etc).
+But even in this case you can, for example, base64 encode encrypted data and store it as a string.
+
+# Type compatibility
+
+When encrypting, Acra takes a bytestring and returns a bytestring.
+So all the data you wanna protect should be transformed into a bytestring:
+1) text strings can be taken "as is"
+2) numbers should be transformed into a string representation\*, then see #1
+3) complex objects should be serialized into a text/binary string, so you got a choice
+   between JSON, BSON, MessagePack and many other formats
+
+\* — be aware that length of encrypted data is not protected, and knowing number of digits
+could give some hint about value, even though it was encrypted;
+to protect from such attacks you can, for example, make all numbers the same length by prepending zero digits
+(like `%010d` for `printf`)
+
+The responsibility of performing these kind of conversions will be on the application.
+
+## One way to solve type problem
+
+Instead of making everything you wanna encrypt a byte string,
+you can use [tokenization]({{< ref "acra/security-controls/tokenization/_index.md" >}}).
+This will help keeping column types as before, but the values won't be stored as plaintext anymore.
```

Comment:
actually, values will be stored as plaintext, but source values will be stored securely as ciphertext. Better to explain that source values will be stored securely and replaced for storage with new random values.